### ISSUE TITLE
catchup history commits since GH104-refactor-using-carter was created

### ISSUE THREAD

**Issue by samatrhea** [RHEAGROUP/COMET-WebServices-Community-Edition#261]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET Web Services
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WebServices-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WebServices-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Update the code using commits performed on development since barnch GH104-refactor-using-carter was created. The following commits are of interest:

- [x] 5ae927baefebe4733ea9d37d7c73ced5b0444de0 - (16/04/2021 10:50:35) - changes already implemented
- [x] 5bbe90fd4d21656cfb949007038f5271baf090ea - (11/06/2021 16:01:06) - changes taken into account
- [x] 6e574531669ecc77389964da51f9844acd723936 - (12/07/2021 08:58:20) - changes already implemented
- [x] 9fbd09dd86c0b7a8edbee3e362276f731bf79a95 - (14/07/2021 08:57:45) - fix associated unit test
- [x] 257dc75f6515a99c9aa085573894540f1e3d933b - (06/08/2021 13:02:06) - changes taken into account
- [x] 988b84eefadc1cedbf9b4f427243d893760576ae - (19/09/2021 18:31:17) - changes already implemented
- [x] 20a59b81b2185be92907a0cf003843ed530cc723 - (23/09/2021 14:44:20) - changes already implemented
- [x] ca933b52f97165b383b1d58ce2c792e60befc324 - (23/09/2021 23:04:27) - changes already implemented
- [x] 57e57db65dac137436f9843de493eaf14a672380 - (27/09/2021 14:07:20) - changes already implemented
- [x] 61779c8fc29809eb14fe5b27595ed2d4eac150b4 - (29/09/2021 12:42:07) - changes already implemented
- [x] 110bd7b4f530fa463eaccef9fa3e76bd6dc8f962 - (04/10/2021 15:26:09) - changes already implemented
- [x] 3c0db904f4bfd39ca1cb1f561ce7f7401372cf42 - (01/11/2021 09:15:14) - changes taken into account
- [ ] e6f7166f7accdb8bcae51ea9abc6459c320c78e2 - (03/01/2022 13:53:01) (IIS implementation)
- [ ] 17a24dfc2a8235222202616a5eb3d6b69231a628 - (03/01/2022 14:23:58) (dependency bump)
- [x] 1a7010786e3470ccc9afc172409a0303db467989 - (23/02/2022 11:51:11) - changes taken into account
- [ ] 058cf5fd4a07b789e86c930dad02b9c99cb6d273 - (23/02/2022 13:12:10)  (dependency bump)
- [x] 973b36f71b24e46e9d5e949d884ccbfd577d6ec0 - (23/02/2022 13:59:32) - changes taken into account
- [ ] b03f139563fa6f80ae2d132e40d218d6dc8db388 - (15/03/2022 12:20:37) (IIS Fix)
- [ ] 1507638062b33927ad8baedc3a659ac9136e4a03 - (16/03/2022 13:55:25) (IIS Fix)
- [x] 83e63eeafd2aa8b2370ce6294330a6913c43960d - (28/03/2022 08:45:42) - changes taken into account 
- [x] acafa78fa5179d83663b3325f04fe5f2eb0f6528 - (05/04/2022 13:18:57) - changes taken into account 
- [x] f6d99a1f67b20846dd8c7c06a0b76a480a9f2c0b - (19/04/2022 18:00:31) - changes taken into account 
- [x] f3a7e6c750f5a1ac650d39bbb828125053468f20 - (22/11/2022 15:38:20) - changes taken into account
- [x] 3d2afede2bc8f443da82ebf06bec7dfedcc5ab77 - (22/12/2022 14:39:00) - changes taken into account
- [x] 07e09d18f5ce0c535bdce75d77c3ee2fb47d5ab0 - (16/06/2023 09:23:33) - changes taken into account
- [x] 7287212efe0eb9f7c818d154b2c6348c409b9818 - (31/07/2023 11:32:38) - changes taken into account
- [x] 7781dcffbcf26534c4d1115551975c155a30304f - (31/08/2023 08:38:33) - changes taken into account
- [x] 1cf94ec0b2628c0ff655b0e0beb33610c01d041d - (01/09/2023 16:55:22) - changes taken into account

If more commits are added, make sure to add them here and implement the changes

This ticket shall only be closed once the net6-carter version is merged onto development and master and is the new version of the COMET Services

### Steps to Reproduce
NA

### System Configuration
NA

**Comment by samatrhea** [comment]
a new issue has been created for IIS support, see #293. this issue can be closed as the new implementation has been merged into the development branch
